### DISCUSSION TITLE
Consideration while deploying yatai-service behind a reverse proxy

### DISCUSSION THREAD

**Discussion by jiyer2016** [bentoml/BentoML#1057] [General] [answered]
When I deploy yatai-service behind a reverse proxy - I envision exposing a single URL with two sub-contexts.

```
https://friendly-url.com/webui -- maps to port 50051
https://friendly-url.com/yatai -- maps to port 3000
```

I notice that in such a setup - calls to `https://friendly-url.com/yatai`  get routed to `127.0.0.1/:3000/webui`  and that doesnt result in the correct page being served.

I am thinking there should be a way to configure a baseURL for both "webui" and "yatai".
For e.g. jupyterhub has this feature - https://jupyterhub.readthedocs.io/en/stable/getting-started/networking-basics.html#adjusting-the-hub-s-url

```
Adjusting the hub’s URL
The hub will most commonly be running on a hostname of its own. If it is not – for example, if the hub is being reverse-proxied and being exposed at a URL such as https://proxy.example.org/jupyter/ – then you will need to tell JupyterHub the base URL of the service. In such a case, it is both necessary and sufficient to set c.JupyterHub.base_url = '/jupyter/' in the configuration.
```

Please let me know if there are alternative ways of getting around this.

**Comment by parano** [comment, marked as answer]
Great suggestion @jiyer2016 - this has come up a few times in the slack channel before. Unfortunately, reverse proxy with a location prefix is not currently supported in BentoML YataiService. I think the GRPC side should just work, but we need to add the URL prefix support in the YataiService's web server and web UI code.

**Reply by jiyer2016** [reply]
Thanks @parano - Closing this item - since it is now being tracked under https://github.com/bentoml/BentoML/issues/1063
